src/core/agent/services/signifyNotificationService.ts

```diff
@@ -158,7 +158,38 @@ class SignifyNotificationService extends AgentService {
     notif: Notification,
     callback: (event: KeriaNotification) => void
   ) {
-    // We only process with the credential and the multisig at the moment
+    if (notif.a.r === NotificationRoute.MultiSigRpy) {
+      const multisigNotification = await this.props.signifyClient
+        .groups()
+        .getRequest(notif.a.d)
+        .catch((error) => {
+          const errorStack = (error as Error).stack as string;
+          const status = errorStack.split("-")[1];
+          if (/404/gi.test(status) && /SignifyClient/gi.test(errorStack)) {
+            return [];
+          } else {
+            throw error;
+          }
+        });
+      if (!multisigNotification || !multisigNotification.length) {
+        await this.markNotification(notif.i);
+        return;
+      }
+      const multisigId = multisigNotification[0]?.exn?.a?.gid;
+      if (!multisigId) {
+        await this.markNotification(notif.i);
+        return;
+      }
+      const multisigIdentifier = await Agent.agent.identifiers.getIdentifier(
+        multisigId
+      );
+      if (!multisigIdentifier) {
+        await this.markNotification(notif.i);
+        return;
+      }
+      await this.markNotification(notif.i);
+      return await Agent.agent.multiSigs.joinAuthorization(notif.a.d);
```

Comment:
> Hmm, can't we just directly call await Agent.agent.multiSigs.joinAuthorization(notif.a.d); and if the notification or multi-sig doesn't exist, it will fail in that function?

Do you mean we can skip the check if this is a valid multisig notification?
If we don't return here, the code below will create some unnecessary records in our storage since we want them to join these requests automatically.

Reply:
Yes, I think `joinAuthorization` will check if it's a valid multi-sig too so one place would be enough? However, as I said in another comment, we may need to differentiate between different multi-sig rpy messages so maybe we need this.

> If we don't return here, the code below will create some unnecessary records in our storage since we want them to join these requests automatically.


I just mean this so the return type is consistent:
```
await Agent.agent.multiSigs.joinAuthorization(notif.a.d);
return;
```

Reply:
I think we can still remove all the checks for `multisigId` and `multisigIdentifier` in this file - it will just fail later which is OK

Reply:
Since we are calling `processNotification` in the `while(true)` loop, it will make some unnecessary crashes in our loop if we do so.

Reply:
Can you explain that more? In what scenario is `multisigIdentifier` `undefined` in one loop of the while loop but not the other?

Reply:
What I mean is: If we don't check and return here, we will have an uncaught error when the `joinAuthorization` got called and it will break the loop.

Reply:
OK, fair point. I've created a ticket to cover this later in general - it's OK to leave it as it is for now.

But we should make sure errors don't break the loop sometime in the future because we may have edge cases we haven't covered for and lose all notifications.